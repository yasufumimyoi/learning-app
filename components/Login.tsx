import { useRouter } from 'next/router'
import { firebase } from '../firebase/config'

const Auth = () => {
  const router = useRouter()

  const handleGuestLogin = () => {
    firebase
      .auth()
      .setPersistence(firebase.auth.Auth.Persistence.SESSION)
      .then(() => {
        router.push('/courses')
        firebase.auth().signInAnonymously()
      })
      .catch(function (error) {
        const errorCode = error.code
        const errorMessage = error.message
        console.log(errorCode, errorMessage)
      })
  }

  return (
    <div className="flex">
      <div className="mr-4">
        <button
          className="bg-green-500 text-white rounded px-8 py-2 focus:outline-none"
          onClick={handleGuestLogin}
        >
          GUEST
        </button>
      </div>
      <div>
        <button
          className="bg-green-500 text-white rounded px-8 py-2 focus:outline-none"
          onClick={() => router.push('/signin')}
        >
          SIGNIN
        </button>
      </div>
    </div>
  )
}

export default Auth
