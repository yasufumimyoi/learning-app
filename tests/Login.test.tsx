import { render } from '@testing-library/react'
import Login from '../components/Login'

it('Loginのsnapshot"', () => {
  const component = render(<Login />)
  expect(component).toMatchSnapshot()
})
