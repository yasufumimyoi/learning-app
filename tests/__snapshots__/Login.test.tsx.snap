// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login„ÅÆsnapshot" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex"
      >
        <div
          class="mr-4"
        >
          <button
            class="bg-green-600 text-white rounded px-8 py-2 focus:outline-none"
          >
            GUEST
          </button>
        </div>
        <div>
          <button
            class="bg-green-600 text-white rounded px-8 py-2 focus:outline-none"
          >
            LOGIN
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex"
    >
      <div
        class="mr-4"
      >
        <button
          class="bg-green-600 text-white rounded px-8 py-2 focus:outline-none"
        >
          GUEST
        </button>
      </div>
      <div>
        <button
          class="bg-green-600 text-white rounded px-8 py-2 focus:outline-none"
        >
          LOGIN
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
